experimental/origin-dapp2/src/pages/user/User.js

```diff
@@ -1,13 +1,29 @@
 import React, { Component } from 'react'
 import { Query } from 'react-apollo'
 import get from 'lodash/get'
+import { fbt } from 'fbt-runtime'
 
 import IdentityQuery from 'queries/Identity'
 import Reviews from 'components/Reviews'
 import Avatar from 'components/Avatar'
 
 import UserListings from './_UserListings'
 
+function parseUserSocialID(attestations) {
+  const socialUrlObject = {}
+  attestations.forEach(attestation => {
+    const socialData = JSON.parse(attestation)
+    if (
+      socialData.data.attestation.site &&
+      socialData.data.attestation.site.userId
+    ) {
+      socialUrlObject[socialData.data.attestation.site.siteName] =
+        socialData.data.attestation.site.userId.raw
```

Comment:
I think you also want to check if socialData.data.attestation.site.userId and socialData.data.attestation.site.userId.raw.

For facebook attestations for example, socialData.data.attestation.site.siteName exists but there is no socialData.data.attestation.site.userId.

For other social sites in the future, socialData.data.attestation.site.userId may existing but not the raw field (for example if user opted for keeping that field encrypted).

Sorry, this is a bit complicated... Here is a doc with some more details on the attestation schema to help you understand: https://docs.google.com/document/d/1PYqe1ggL60xCQqX7C2gj0qZZeEsI0aqcd7cWvtenVuc/edit?usp=sharing

Reply:
In particular note the schema for an Attribute. Can have either a raw or hash field.

Reply:
I will review the google doc schema. No worries on the complication, I want to be sure it's going to work for the project correctly.